### DISCUSSION TITLE
guide/code/features/

### DISCUSSION THREAD

**Discussion by giscus** [pengzhanbo/vuepress-theme-plume#165] [docs-comment]
# guide/code/features/

主题在代码高亮功能上，进一步支持了更多的特性，它们能够帮助你的代码块更加具备表达力。 代码行号 主题默认显示代码行号，它通过 plugins.shiki.line-numbers 来控制。 你还可以通过 :line-numbers / :no-line-numbers 来控制当前代码块是否显示代码行号。 还可以通过在 :line-numbers 之后添...

https://theme-plume.vuejs.press/guide/code/features/

**Comment by bawanchun** [comment]
博主你好，再弱弱的问一句，比如某行代码太长了，自动换行该怎么设置？

**Reply by pengzhanbo** [reply]
代码块中的 每一行都是强制不换行的，自动换行 会导致 界面上的 代码行数 不能对齐。 当前 主题还没有找到一种 兼容性较好的 方式来实现 自动换行 和 代码行数 自动对齐。因此出于视觉考虑，代码块中的 代码行都是强制不换行。 因此建议结合 代码语法、表达式 等具体情况，手动进行换行。

**Comment by Nick110** [comment]
我的JavaScript代码块不带颜色是为什么？

**Reply by pengzhanbo** [reply]
请[查看 代码块 文档](https://theme-plume.vuejs.press/guide/code/intro/#%E6%A6%82%E8%BF%B0)

**Comment by dingyuqi** [comment]
```
        shiki: {
            languages: ["sql", "shell", "mermaid", "go", "html", "php", "python", "bash", "c++", "js", "css", "yaml", "md"],
            whitespace: true,
        },
```
这个配置似乎无法生效.

**Reply by dingyuqi** [reply]
添加了配置之后, 每行代码的前面没有对空格和tab进行响应的渲染.
